src/main/kotlin/uk/gov/justice/digital/hmpps/welcometoprison/model/prison/transfers/TransfersService.kt

```diff
@@ -22,27 +24,52 @@ class TransfersService(
   private val arrivalListener: ArrivalListener,
 ) {
 
-  fun getTransfer(agencyId: String, prisonNumber: String): Transfer =
-    getTransfers(agencyId).find { it.prisonNumber == prisonNumber }
-      ?: throw NotFoundException("Could not find transfer with prisonNumber: '$prisonNumber'")
+  fun getTransfer(agencyId: String, prisonNumber: String): TransferWithMainOffence {
+    val offenderMovementWithPnc = getOffenderMovementsWithPnc(agencyId, prisonNumber)[0]
+    var mainOffence = prisonApiClient.getMainOffence(offenderMovementWithPnc.offenderMovement.bookingId)
+    return TransferWithMainOffence(
+      firstName = Name.properCase(offenderMovementWithPnc.offenderMovement.firstName),
+      lastName = Name.properCase(offenderMovementWithPnc.offenderMovement.lastName),
+      dateOfBirth = offenderMovementWithPnc.offenderMovement.dateOfBirth,
+      fromLocation = offenderMovementWithPnc.offenderMovement.fromAgencyDescription,
+      prisonNumber = offenderMovementWithPnc.offenderMovement.offenderNo,
+      date = offenderMovementWithPnc.offenderMovement.movementDate,
+      pncNumber = offenderMovementWithPnc.pncNumber,
+      mainOffence = mainOffence?.offenceDescription,
+    )
+  }
 
-  fun getTransfers(agencyId: String): List<Transfer> {
-    val transfers = prisonApiClient.getPrisonTransfersEnRoute(agencyId, LocalDate.now())
+  fun getOffenderMovementsWithPnc(agencyId: String, forPrisonNumber: String?): List<OffenderMovementWithPnc> {
```

Comment:
Just wondering whether it would be more direct to get the bookingId from prison service using api/offenders/{offenderNo}. ??